domain/src/main/java/org/oppia/android/domain/translation/TranslationController.kt

```diff
@@ -66,24 +72,23 @@ class TranslationController @Inject constructor(
   private val dataProviders: DataProviders,
   private val localeController: LocaleController,
   private val asyncDataSubscriptionManager: AsyncDataSubscriptionManager,
-  private val machineLocale: OppiaLocale.MachineLocale
+  private val machineLocale: OppiaLocale.MachineLocale,
+  private val languageConfigRetriever: LanguageConfigRetriever,
+  private val cacheStoreFactory: PersistentCacheStore.Factory,
+  private val oppiaLogger: OppiaLogger,
 ) {
-  // TODO(#52): Finish this implementation. The implementation below doesn't actually save/restore
-  //  settings from the local filesystem since the UI has been currently disabled as part of #20.
-  //  Also, there should be a proper default locale for pre-profile selection (either a default
-  //  app-wide setting determined by the administrator, or the last locale used by a profile)--more
-  //  product & UX thought is needed here. Further, extra work is likely needed to handle the case
-  //  when the user switches between a system & non-system language since the existing system
-  //  defaulting behavior breaks down when the system locale is overwritten (such as when a
-  //  different language is selected).
+  // TODO(#4627): per profile audio language slection part not covered in this PR but should be when this to do is updated.
```

Comment:
TODOs should be written with an actionable outcome so that others can follow along. "This PR" isn't context that makes sense in the context of just the code (which is what everyone else would see when encountering this TODO). Please revise. I suggest looking at other TODOs in the codebase to get an idea on how to write a clear one.

Reply:
Updated, although now referring to #4938 which might fix per profile audio language selection.  #4627 is closed and replaced with #4938.